autogpt_platform/backend/backend/sdk/registry.py

```diff
@@ -63,27 +64,35 @@ def register_provider(cls, provider: "Provider") -> None:
 
             # Register OAuth handler if provided
             if provider.oauth_config:
-                # Dynamically set PROVIDER_NAME if not already set
-                if (
-                    not hasattr(provider.oauth_config.oauth_handler, "PROVIDER_NAME")
-                    or provider.oauth_config.oauth_handler.PROVIDER_NAME is None
-                ):
-                    # Import ProviderName to create dynamic enum value
-                    from backend.integrations.providers import ProviderName
+                oauth_env_vars_set = os.getenv(
+                    provider.oauth_config.client_id_env_var
+                ) and os.getenv(provider.oauth_config.client_secret_env_var)
```

Comment:
unnecessary nested `if` can be replaced with one (and avoid the indent change):
```suggestion
            if (
                provider.oauth_config
                and os.getenv(provider.oauth_config.client_id_env_var)
                and os.getenv(provider.oauth_config.client_secret_env_var)
            ):
```